api/src/main/java/com/codeforcommunity/exceptions/TeamLeaderOnlyRouteException.java

```diff
@@ -0,0 +1,22 @@
+package com.codeforcommunity.exceptions;
+
+import com.codeforcommunity.rest.FailureHandler;
+import io.vertx.ext.web.RoutingContext;
+
+public class TeamLeaderOnlyRouteException extends HandledException {
```

Comment:
Maybe call this wrong TeamRoleException and take a TeamRole value(s) so that we can reuse it down the line

Reply:
Done